### PR TITLE
fix(api): resolve access control and security issues in release artifact endpoints

### PR DESCRIPTION
**Pull request by vpetersson**
BREAKING CHANGE: Error messages no longer reveal team/resource existence details

Security:
- Fix information disclosure vulnerability in error messages
- Replace team-specific errors with generic "Access denied" to prevent resource enumeration attacks

Access Control:
- Remove redundant _get_user_team_id() checks that broke API token auth
- Fix 403 Forbidden when tagging SBOMs with releases via GitHub Actions
- Affected endpoints: add/remove artifacts, list releases, document/SBOM tagging

API Improvements:
- Return 409 Conflict for duplicate artifacts (RESTful best practice)
- Add missing error_code to all 403/404 responses for client parsing
- Add DUPLICATE_ARTIFACT error code to schema

Testing:
- Add comprehensive tests for API token authentication
- Add tests for 409 Conflict behavior on duplicate artifacts
- Update assertions to verify generic error messages

Fixes: 403 Forbidden on release tagging after SBOM upload in CI/CD

### CODE REVIEW

File: sbomify/apps/core/apis.py [hunk 1]
```diff
@@ -665,7 +665,7 @@ def patch_product(request: HttpRequest, product_id: str, payload: ProductPatchSc
                 # Verify all projects exist and belong to the same team
                 projects = Project.objects.filter(id__in=project_ids, team_id=product.team_id)
                 if len(projects) != len(project_ids):
-                    return 400, {"detail": "Some projects were not found or don't belong to this team"}
+                    return 400, {"detail": "Some projects were not found or inaccessible"}
```
[the remark this answers is not in the document]
Comment (vpetersson) [reply sbomify/apps/core/apis.py]: Ignored.
